packages/astro/src/runtime/server/render/astro/render-template.ts

```diff
@@ -36,23 +36,15 @@ export class RenderTemplateResult {
 	async *[Symbol.asyncIterator]() {
 		const { htmlParts, expressions } = this;
 
-		let iterables: Array<EagerAsyncIterableIterator> = [];
-		// all async iterators start running in non-buffered mode to avoid useless caching
-		for (let i = 0; i < htmlParts.length; i++) {
-			iterables.push(new EagerAsyncIterableIterator(renderChild(expressions[i])));
-		}
-		// once the execution of the next for loop is suspended due to an async component,
-		// this timeout triggers and we start buffering the other iterators
-		setTimeout(() => {
-			// buffer all iterators that haven't started yet
-			iterables.forEach((it) => !it.isStarted() && it.buffer());
-		}, 0);
+		let iterables = bufferIterators(expressions.map((e) => renderChild(e)));
 		for (let i = 0; i < htmlParts.length; i++) {
 			const html = htmlParts[i];
 			const iterable = iterables[i];
 
 			yield markHTMLString(html);
-			yield* iterable;
+			if (iterable) {
+				yield* iterable;
+			}
```

Comment:
Is there a reason this is wrapped with an `if`?

Reply:
I'm glad you ask :-)
There might be more `htmlParts` than `expressions`, and so far it didn't matter because apparently `renderChild` doesn't mind undefined values. However, if we map the `expressions` based on their length, and not the length of `htmlElements` as before, we will end up with less entries, and while we can pass `undefined` to `renderChild`, it will crash if we `yield* undefined`. That's why we need an additional check here.

Reply:
`renderChild(undefined)` will `yield undefined` (TIL you can `await undefined`), so that's why it worked before. Do you think my change will improve the readability?